test/performance/odp_ipsec.c

```diff
@@ -722,56 +763,72 @@ run_measure_one_async(ipsec_args_t *cargs,
 
 	fill_time_record(start);
 
-	while ((packets_sent < cargs->iteration_count) ||
-	       (packets_received < cargs->iteration_count)) {
+	while ((packets_sent < cargs->packet_count) ||
+	       (packets_received < cargs->packet_count)) {
 		odp_event_t ev;
 
-		if ((packets_sent < cargs->iteration_count) &&
-		    (packets_sent - packets_received <
-		     cargs->in_flight)) {
-			pkt = make_packet(pkt_pool, payload_length);
-			if (ODP_PACKET_INVALID == pkt)
+		num_pkts = cargs->packet_count - packets_sent;
+
+		/* Enqueue up to burst size */
+		num_pkts = num_pkts > burst_size ? burst_size : num_pkts;
+
+		/* Enqueue up to (max in flight - current in flight) */
+		in_flight = packets_sent - packets_received;
+		pkts_allowed = max_in_flight - in_flight;
+		num_pkts = num_pkts > pkts_allowed ? pkts_allowed : num_pkts;
+
+		if (odp_likely(num_pkts)) {
+			if (odp_unlikely(make_packet_multi(pkt_pool,
+							   payload_length,
+							   pkt,
+							   num_pkts)))
 				return -1;
 
-			if (cargs->debug_packets)
-				odp_packet_print_data(pkt, 0,
-						      odp_packet_len(pkt));
+			if (cargs->debug_packets) {
+				for (i = 0; i < num_pkts; i++)
+					odp_packet_print_data(pkt[i], 0,
+							      odp_packet_len(pkt[i]));
+			}
 
-			rc = odp_ipsec_out_enq(&pkt, 1,
+			rc = odp_ipsec_out_enq(pkt, num_pkts,
```

Comment:
There is a packet leak also here if the enqueue does not consume all input packets.

Reply:
Agreed. Will fix in v3.